openreview/venue/venue.py

```diff
@@ -1088,10 +1088,21 @@ def ithenticate_create_and_upload_submission(self):
                 owner = (
                     submission.signatures[0]
                     if submission.signatures[0].startswith("~")
-                    else submission.content["authorids"]["value"][0]
+                    else self.client.get_note_edits(note_id=submission.id, invitation=self.get_submission_id(), sort='tcdate:asc')[0].signatures[0]
                 )
+                print(f"Creating submission for {submission.id} with owner {owner}")
                 owner_profile = self.client.get_profile(owner)
 
+                eula_version = submission.content.get("iThenticate_agreement", {}).get("value", "v1beta").split(":")[-1].strip()
+                
+                iThenticate_client.accept_EULA(
+                    user_id=owner_profile.id,
+                    eula_version=eula_version,
+                    timestamp=datetime.datetime.fromtimestamp(
```

Comment:
this timestamp is computed twice, can you save it in variable and use it for both calls?